### PR TITLE
SOC-983 password toggler for mercury login

### PR DESCRIPTION
**Pull request by qaga**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java [hunk 1]
```diff
@@ -123,4 +126,19 @@ public void clickOnRegisterLink(){
   public String getLoginHeaderText(){
     return loginHeader.getText();
   }
+
+  public void typePassword(String password) { passwordField.sendKeys(password); }
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java]: MAJOR Move this closing curly brace to the next line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java [hunk 2]
```diff
@@ -123,4 +126,19 @@ public void clickOnRegisterLink(){
...
+
+  public void clickOnPasswordToggler() { passwordToggler.click(); };
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java]: MAJOR Move this closing curly brace to the next line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)
MAJOR Remove this empty statement. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java [hunk 3]
```diff
@@ -123,4 +126,19 @@ public void clickOnRegisterLink(){
...
+
+  public Boolean isPasswordTogglerDisabled() {
+    String togglerDisabled = passwordField.getAttribute("type");
+    return togglerDisabled.equals("password");
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java]: MAJOR Move the "password" string literal on the left side of this string comparison. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java [hunk 4]
```diff
@@ -123,4 +126,19 @@ public void clickOnRegisterLink(){
...
+  }
+
+  public Boolean isPasswordTogglerEnabled() {
+    String togglerDisabled = passwordField.getAttribute("type");
+    return togglerDisabled.equals("text");
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java]: MAJOR Move the "text" string literal on the left side of this string comparison. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues:
* MAJOR 7 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/19195d123b95841e682b94dd233f865a7dbe2aba/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java#L107) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)
* MAJOR [Either log or rethrow this exception.](https://github.com/Wikia/selenium-tests/blob/19195d123b95841e682b94dd233f865a7dbe2aba/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java#L66) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java (hunk 1 above)
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java]: please don't collapse blocks, use:
```
public void typePassword(String password)
{
 passwordField.sendKeys(password);
}
```

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java (hunk 2 above)
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java]: the same as above

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java [hunk 5]
```diff
@@ -123,4 +126,19 @@ public void clickOnRegisterLink(){
   public String getLoginHeaderText(){
     return loginHeader.getText();
   }
+
+  public void typePassword(String password) { passwordField.sendKeys(password); }
+
+  public void clickOnPasswordToggler() { passwordToggler.click(); };
+
+  public Boolean isPasswordTogglerDisabled() {
+    String togglerDisabled = passwordField.getAttribute("type");
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java]: `getAttribute()` can return `null`, you should handle that

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java [hunk 6]
```diff
@@ -123,4 +126,19 @@ public void clickOnRegisterLink(){
...
+    return togglerDisabled.equals("password");
+  }
+
+  public Boolean isPasswordTogglerEnabled() {
+    String togglerDisabled = passwordField.getAttribute("type");
```
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/LoginPage.java]: the same as above

**Comment by ludwikkazmierczak** [comment]
LGTM

**Pull request merged**